### PR TITLE
use RTC as system clock for CC13X2/CC26X2

### PR DESCRIPTION
**Pull request by vanti**
Add RTC timer driver for CC13X2/CC26X2, and use it instead of systick
as system clock. It is necessary to use this timer for power
management support, so that the system can exit from deep sleep upon
expiry of timeouts.

A couple of tests also need to be adjusted as a result of using this timer.

### CODE REVIEW

**Review by andyross — APPROVED** [review]
Driver looks pretty good to me.  One nitpick about initialization state, which is fine as long as the reset process guarantees the counter value.

**Comment by vanti** [comment]
recheck

File: drivers/timer/cc13x2_cc26x2_rtc_timer.c [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	AONRTCEnable();
+
+	irq_unlock(key);
+}
+
+int z_clock_driver_init(struct device *device)
+{
+	ARG_UNUSED(device);
+
+	rtc_last = 0U;
```
[the remark this answers is not in the document]
Comment (vanti) [reply drivers/timer/cc13x2_cc26x2_rtc_timer.c]: That is a good point. Yes, the counter is reset to 0 by initDevice().

**Comment by vanti** [comment]
@galak Not sure why the checks are not finishing...

**Comment by vanti** [comment]
Thanks for the help!

**Pull request merged**